tests/robustness/README.md

```diff
@@ -0,0 +1,71 @@
+# etcd Robustness Testing
+
+Purpose of etcd robustness tests is to validate that etcd upholds
+[API guarantees] and [watch guarantees] under any condition or failure.
+
+Robustness tests achieve that comparing etcd cluster behavior against a simplified model.
+Multiple test encompass different etcd cluster setups, client traffic types and failures experienced by cluster.
+During a single test we create a cluster and inject failures while sending and recording client traffic.
+Correctness is validated by running collected history of client operations against the etcd model and a set of validators.
+Upon failure tests generate a report that can be used to attribute whether failure was caused by bug in etcd or test framework. 
+
+[API guarantees]: https://etcd.io/docs/latest/learning/api_guarantees/
+[watch guarantees]: https://etcd.io/docs/latest/learning/api/#watch-streams
+
+## Running locally 
+
+1. Build etcd with failpoints
+    ```bash
+    make gofail-enable
+    make build
+    make gofail-disable
+    ```
+2. Run the tests
+
+    ```bash
+    make test-robustness
+    ```
+   
+    Optionally you can pass environment variables:
+    * `GO_TEST_FLAGS` - to pass additional arguments to `go test`. 
+      It is recommended to run tests multiple times with failfast enabled. this can be done by setting `GO_TEST_FLAGS='--count=100 --failfast'`.
+    * `EXPECT_DEBUG=true` - to get logs from the cluster.
+    * `RESULTS_DIR` - to change location where results report will be saved.
```

Comment:
What's the default for RESULTS_DIR ? within the test's tmp directory that is cleaned after success ?

Reply:
No, robustness tests specifically create its own directory in tmp that is not created via `t.TmpDir()`